src/main/kotlin/com/redhat/devtools/intellij/kubernetes/model/client/ClientAdapter.kt

```diff
@@ -52,17 +56,33 @@ open class KubeClientAdapter(client: KubernetesClient) :
     override fun isOpenShift(): Boolean {
         return false
     }
+
+    override fun toOpenShift(): OSClientAdapter {
+            val kubeClient = get()
+            val osClient = kubeClient.adapt(NamespacedOpenShiftClient::class.java)
+            return OSClientAdapter(osClient, kubeClient)
+    }
 }
 
 abstract class ClientAdapter<C : KubernetesClient>(private val fabric8Client: C) {
 
     companion object Factory {
 
+        const val TIMEOUT_CONNECTION = 5000
+        const val TIMEOUT_REQUEST = 5000
+        const val LIMIT_RECONNECT = 2
+
         fun create(
             namespace: String? = null,
             context: String? = null,
             clientBuilder: KubernetesClientBuilder? = null,
-            createConfig: (context: String?) -> Config = { context -> Config.autoConfigure(context) },
+            createConfig: (context: String?) -> Config = { context ->
+                val config = Config.autoConfigure(context)
+                config.connectionTimeout = TIMEOUT_CONNECTION
+                config.requestTimeout = TIMEOUT_REQUEST
+                config.watchReconnectLimit = LIMIT_RECONNECT
+                config
```

Comment:
This is an aspect of the fix: impose timeouts for connecting, requesting and reconnecting.